### PR TITLE
feature: initial release

### PR DESCRIPTION
**Pull request by marcoocram**

### CODE REVIEW

File: src/CustomCollection.php [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+     *
+     * @param mixed $key
+     * @return void
+     */
+    public function offsetUnset($key): void
+    {
+        unset($this->items[$key]);
+    }
+
+    public function count(): int
```
Comment (ggcasuso) [inline src/CustomCollection.php:89]: should you add the docblock? We have it in all class methods

Comment (marcoocram) [reply src/CustomCollection.php:89]: you're right! forgot it!

File: tests/CustomCollectionTest.php [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+use Exception;
+use Letsgoi\CustomCollection\CustomCollection;
+use Letsgoi\CustomCollection\Exceptions\CustomCollectionKeyNotExistException;
+use Letsgoi\CustomCollection\Exceptions\CustomCollectionTypeErrorException;
+use stdClass;
+
+class CustomCollectionTest extends TestCase
+{
+    /** @test */
+    public function it_should_iterate_all_his_items()
```
Comment (adri-navarro) [inline tests/CustomCollectionTest.php:14]: `its_items`?

File: tests/CustomCollectionTest.php [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                return 'string';
+            }
+        };
+
+        $this->assertSame('item', $stringCollection->get(0));
+        $this->assertSame('anotherItem', $stringCollection->get(1));
+    }
+
+    /** @test */
+    public function it_should_throw_exception_on_get_unknown_key_on_collection()
```
Comment (adri-navarro) [inline tests/CustomCollectionTest.php:62]: `exception_when_get`?

**Pull request merged**